### DISCUSSION TITLE
Heavy bot traffic to our API with node-fetch user agent? Friend or foe?

### DISCUSSION THREAD

**Discussion by mlstubblefield** [vercel/community#168] [Help] [answered]
We've started to just recently incrementally adopt next.js hosted on vercel. Not too long after a slow roll out of the site, we noticed a lot of traffic hitting our API (close to 1M additional requests per day).
The API is used for a number of things and our current logging isn't sufficient enough to identify exactly what exactly is being requested.
That being said, one of my first thoughts was "I wonder if it could be Incremental Static Regeneration".
The traffic we're seeing has a couple of qualities that are interesting and led me to this belief
- The user agent is `node-fetch/1.0%20(+https://github.com/bitinn/node-fetch)` (I notice that vercel references various versions of node-fetch-- although I'm not sure if any of them would be the old version with this user agent.
- The IP addresses are super distributed and from AWS ranges.

**Comment by kristjanmar** [comment]
I've been getting this too. Occasionally I get a flood of request to my backend API from AWS IP addresses that overload my PHP server. I'm also using ISR.

**Comment by mlstubblefield** [comment, marked as answer]
This is a little embarrassing, but at the end of the day this was a fetch call that we were making inside of the getStaticProps method. I was able to identify that by setting a custom user agent in that method. Then my logs (unsurprisingly) switched to have the new user agent in them.
